DEPENDENCY_PINNING.md

```diff
@@ -0,0 +1,33 @@
+## Dependency Pinning
+
+Based on the initial discussions from [this discussion thread](https://github.com/nodejs/nodejs.org/discussions/5491), we've decided to use a more strict strategy for handling NPM dependencies within the Node.js Website.
+
+The intent here is to avoid the build process, or the website itself, from breaking due to changes in dependencies. As some dependencies do not respect semantic versioning, this is a real concern. Pinning dependencies also ensures that we stay fixed on a specific version of a dependency; For security updates, Dependabot is still configured to give us security alerts when certain dependencies got security advisories.
+
+### When adding dependencies
+
+The following recommendations are in order when adding a new dependency:
+
+- A dependency should be a `dependencies` if it is part of the build process of the Website or used within runtime code.
+  - Some non-code dependencies are required for the bootstrap of the repository. They either are used on basic build scripts or Git Hooks. Examples include `husky`, `lint-staged` and others.
```

Comment:
non-code is tripping me up here. it's all code. `husky` and `lint-staged` should be devDependencies. They are only concerns for someone running the website locally.

The way I think about it often is, "what does the production bundle need?" vs "what does a developer need?"

Reply:
> non-code is tripping me up here. it's all code. husky and lint-staged should be devDependencies. They are only concerns for someone running the website locally.

Yes and no, if you simply git clone and run `nopm i --omit=dev` it will fail because husky is not there.

I make the line of a devDependency being a dependency used for actual development. Whereas regular `dependencies` are either runtime dependencies or the minimal set of dependencies if someone clones or builds the project. Or any dependency required for the build to succeed, at all. Because the build itself is part of the "production" environment, after all. Even more since we have a mixed SSR fashion, a few dependencies that you consider being "dev" would actually be part of the "runtime" on "edge-functions/SSR"